apps/backend/src/parse/parse.controller.ts

```diff
@@ -61,4 +75,218 @@ export class ParseController {
       fileContents: contents,
     });
   }
+
+  @Post('upload-file')
+  @ApiConsumes('multipart/form-data')
+  @ApiBody({
+    schema: {
+      type: 'object',
+      properties: {
+        fileName: {
+          type: 'string',
+          nullable: false,
+        },
+        fileType: {
+          type: 'string',
+          enum: ['TDI17', 'TDI34', 'SBC_SALES'],
+          nullable: false,
+        },
+        program: {
+          type: 'string',
+          enum: ['SBC', 'LABOUR'],
+          nullable: false,
+        },
+        file: {
+          type: 'string',
+          format: 'binary',
+        },
+      },
+    },
+  })
+  @UseInterceptors(FileInterceptor('file'))
+  async uploadAndParseFile(
+    @Body() body: { fileName: string; fileType: FileTypes; program: string },
+    @UploadedFile() file: Express.Multer.File
+  ) {
+    const { fileName, fileType, program } = body;
+    this.appLogger.log(`Parsing ${fileName} - ${fileType}`);
+    const contents = file.buffer.toString();
+
+    const allFiles = await this.parseService.getAllFiles();
+    const allFilenames = new Set(allFiles.map((f) => f.sourceFileName));
+    if (allFilenames.has(fileName)) {
+      throw new BadRequestException({
+        message: 'Invalid filename, this already exists',
```

Comment:
I'm assuming it's very intentional we aren't enforcing this at the DB level, in order to potentially support file replacement later?

Reply:
Correct! Didn't want to make this migration immediately pending any discussions we might have on updating files